File: CNC modeling 2.0/src/main/java/com/sergey/pisarev/model/ContextMenuCodeArea.java
package com.sergey.pisarev.model;

import javafx.scene.control.ContextMenu;
import javafx.scene.control.MenuItem;
import javafx.scene.input.Clipboard;
import org.fxmisc.richtext.CodeArea;

public class ContextMenuCodeArea {

    public static void installContextMenu(ContextMenu contextMenu, CodeArea codeAreaProgram){
        MenuItem menuItemCopy = new MenuItem("Copy");
        MenuItem menuItemPaste = new MenuItem("Paste");
        MenuItem menuItemCut = new MenuItem("Cut");
        menuItemCopy.setOnAction(event -> codeAreaProgram.copy());
        menuItemPaste.setOnAction(event -> codeAreaProgram.paste());
        menuItemCut.setOnAction(event -> codeAreaProgram.cut());
        contextMenu.getItems().addAll(menuItemCut, menuItemCopy, menuItemPaste);
        codeAreaProgram.setOnContextMenuRequested(event -> {
            Clipboard clipboard = Clipboard.getSystemClipboard();
            String text = clipboard.getString();
            menuItemPaste.setDisable(text == null);
            contextMenu.show(codeAreaProgram, event.getScreenX(), event.getScreenY());
            event.consume();
        });
    }
}

File: CNC modeling 2.0/src/main/java/com/sergey/pisarev/model/ContextMenuCodeArea.kt
package com.sergey.pisarev.model

import javafx.event.EventHandler
import javafx.scene.control.ContextMenu
import javafx.scene.control.MenuItem
import javafx.scene.input.Clipboard
import javafx.scene.input.ContextMenuEvent
import org.fxmisc.richtext.CodeArea

object ContextMenuCodeArea {

    fun installContextMenu(contextMenu: ContextMenu, codeAreaProgram: CodeArea) {
        val menuItemCopy = MenuItem("Copy")
        val menuItemPaste = MenuItem("Paste")
        val menuItemCut = MenuItem("Cut")
        menuItemCopy.onAction = EventHandler { codeAreaProgram.copy() }
        menuItemPaste.onAction = EventHandler { codeAreaProgram.paste() }
        menuItemCut.onAction = EventHandler { codeAreaProgram.cut() }
        contextMenu.items.addAll(menuItemCut, menuItemCopy, menuItemPaste)
        codeAreaProgram.onContextMenuRequested = EventHandler { event: ContextMenuEvent ->
            val clipboard = Clipboard.getSystemClipboard()
            val text = clipboard.string
            menuItemPaste.isDisable = text == null
            contextMenu.show(codeAreaProgram, event.screenX, event.screenY)
            event.consume()
        }
    }
}